examples/validation/validate_line_loop.py

```diff
@@ -0,0 +1,119 @@
+"""
+Line loop
+=========
+
+Drawing multiple line loops using material.loop and separating line-pieces with nans.
+"""
+
+# ruff: noqa: RUF005
+
+# sphinx_gallery_pygfx_docs = 'screenshot'
+# sphinx_gallery_pygfx_test = 'compare'
+
+import numpy as np
+from rendercanvas.auto import RenderCanvas, loop
+import pygfx as gfx
+
+
+canvas = RenderCanvas(size=(1000, 800))
+renderer = gfx.WgpuRenderer(canvas)
+scene = gfx.Scene()
+
+
+def circle(n, x=0, y=0):
+    t = np.linspace(0, 2 * np.pi, n, endpoint=False)
+    return np.stack([x + np.sin(t), y + np.cos(t), np.zeros_like(t)], axis=1)
+
+
+nanpoint = np.full((1, 3), np.nan, dtype="f4")
+rect_points = np.array([[-1, -1, 0], [-1, +1, 0], [+1, +1, 0], [+1, -1, 0]], "f4")
+
+# This line-piece has all straight corners. It covers the fix for a
+# bug wherer 90 degree corner, aligned with the coordinate frame,
+# resulted in a glitch in the line join.
+all_corners = np.array(
+    [
+        [0, 0, 0],
+        [0, 1, 0],
+        [1, 1, 0],
+        [1, 0, 0],
+        [0.8, 0, 0],
+        [0.8, 0.8, 0],
+        [0.2, 0.8, 0],
+        [0.2, 0.2, 0],
+        [0.4, 0.2, 0],
+        [0.4, 0.6, 0],
+        [0.6, 0.6, 0],
+        [0.6, 0, 0],
+    ],
+    "f4",
+)
+all_corners *= 5
+all_corners += (-8, 3, 0)
+
+
+positions = np.vstack(
+    [
+        # Special case
+        all_corners,
+        nanpoint,
+        # Increasingly more corners
+        circle(0, -7, 0),
+        nanpoint,
```

Comment:
```suggestion
        nanpoint,
        nanpoint,
```

ensure successive nanpoints don't crash the shader